collectionService.js

```diff
@@ -410,7 +414,32 @@ var CollectionService = module.exports = BaseService.extend( {
 			this.length++;
 		}
 
-		_.extend( this._recordsById[ recordId ], dto );
+		for( field in dto ) {
+			if( _.isObject( dto[ field ] ) ){
+				this._recordsById[ recordId ][ field ] = _.clone( dto[ field ] );
+				this._deepFreeze( this._recordsById[ recordId ][ field ] );
+			} else this._recordsById[ recordId ][ field ] = dto[ field ];			
+		}
+	},
+
+	// From : https://github.com/jsdf/deep-freeze
+	_deepFreeze( obj ) {
+		Object.freeze( obj );
+	
+		var objIsFunction = typeof obj === "function";
```

Comment:
baseline properties will never be functions, let's get rid of this complexity since we got performance to tune here

Reply:
got it! gone